src/pages/GetStarted/CustomWorkspaceTab/StorageType.tsx

```diff
@@ -0,0 +1,184 @@
+ * Copyright (c) 2018-2020 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import React from 'react';
+import {
+  FormGroup,
+  Button,
+  Modal,
+  ModalVariant,
+  Select,
+  SelectOption,
+  SelectVariant,
+  SelectDirection,
+  TextContent,
+  Text
+} from '@patternfly/react-core';
+import { connect } from 'react-redux';
+import { AppState } from '../../../store';
+import * as WorkspaceStore from '../../../store/Workspaces';
+import * as BrandingStore from '../../../store/Branding';
+
+import styles from './StorageType.module.css';
+
+export enum StorageType {
+  async = 'Asynchronous',
+  ephemeral = 'Ephemeral',
+  persistent = 'Persistent',
+}
+
+type Props = {
+  storageType?: StorageType;
+  onChange?: (storageType: StorageType) => void;
+} & {
+  brandingStore: BrandingStore.State;
+  workspaces: WorkspaceStore.WorkspacesState,
+}
+type State = {
+  isOpen?: boolean;
+  selected?: string;
+  isModalOpen?: boolean;
+};
+
+export class StorageTypeFormGroup extends React.PureComponent<Props, State> {
+  storageTypes: StorageType[] = [];
+  preferredType: string;
+  options: any[] = [];
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isOpen: false,
+      isModalOpen: false,
+    };
+
+    const settings = this.props.workspaces.settings;
+    if (settings) {
+      const available_types = settings['che.workspace.storage.available_types'];
+      if (available_types) {
+        this.storageTypes = available_types.split(',') as StorageType[];
+        this.preferredType = settings['che.workspace.storage.preferred_type'];
+        this.storageTypes.forEach(type => {
+          this.options.push({ value: StorageType[type] });
+        });
+      }
+    }
+  }
+
+  public componentDidUpdate(prevProps: Props): void {
+    if (prevProps.storageType !== this.props.storageType) {
+      const selected = this.props.storageType;
+      this.setState({ selected });
+    }
+  }
+
+  public componentDidMount(): void {
+    const selected = this.props.storageType ? this.props.storageType : this.preferredType;
+    this.setState({ selected });
+  }
+
+  private handleToggle(isOpen: boolean): void {
+    this.setState({ isOpen });
+  }
+
+  private handleSelect(event, selection): void {
+    if (this.props.onChange) {
+      this.props.onChange(selection);
+    }
+    this.setState({
+      selected: selection,
+      isOpen: false
+    });
+  }
+
+  private handleModalToggle(): void {
+    this.setState(({ isModalOpen }) => ({
+      isModalOpen: !isModalOpen
+    }));
+  }
+
+  private getModalContent(): React.ReactNode {
+    const hasAsync = this.storageTypes.some(type => StorageType[type] === StorageType.async);
+    const getAsync = hasAsync ?
```

Comment:
maybe rename `getAsync` to `asyncTypeDescr`?

Reply:
done